python/lsst/meas/astrom/match_probabilistic_task.py

```diff
@@ -124,20 +129,28 @@ def match(
 
         Returns
         -------
-        catalog_out_ref : `pandas.DataFrame`
+        catalog_out_ref : `pandas.DataFrame` | `astropy.table.Table`
             Reference matched catalog with indices of target matches.
-        catalog_out_target : `pandas.DataFrame`
+        catalog_out_target : `pandas.DataFrame` | `astropy.table.Table`
             Reference matched catalog with indices of target matches.
         """
         if logger is None:
             logger = self.log
 
+        is_ref_pd = isinstance(catalog_ref, pd.DataFrame)
+        is_target_pd = isinstance(catalog_target, pd.DataFrame)
+        if is_ref_pd or is_target_pd:
+            logger.warning("pandas DataFrame inputs are deprecated")
+
         config = self.config
 
         if config.column_ref_order is None:
-            flux_tot = np.nansum(
-                catalog_ref.loc[:, config.columns_ref_flux].values, axis=1
+            fluxes = (
+                catalog_ref.loc[:, config.columns_ref_flux].values
+                if is_ref_pd else
+                [catalog_ref[key] for key in config.columns_ref_flux]
             )
+            flux_tot = np.nansum(fluxes, axis=1 if is_ref_pd else 0)
```

Comment:
Is this needed because the pandas selection leaves an extra dimension of length `1`?  If this is the case would `np.squeeze()` help in the pandas case above?